### PR TITLE
Skip setting UID for logged in users

### PR DESCRIPTION
**Pull request by martbhell**
Use a list of logged in users with the unix "users" command
Also for the usermod commands which use both .group and .groups variables make sure that those are defined

### CODE REVIEW

File: tasks/main.yml [hunk 1]
```diff
@@ -26,6 +26,16 @@
     when:
       - ansible_ssh_user is undefined
 
+    # This list includes the user used to ssh into the system
+  - name: grab a space separated string of users with the users linux command
```
Comment (oscarkraemer) [inline tasks/main.yml]: Would it make sense to have a 'changed_when: False' here?

File: tasks/main.yml [hunk 2]
```diff
@@ -90,32 +100,33 @@
         loop_var: user
         label: "{{ user.name }} {{ user.state }}"
       when:
-        # Ansible user module fails to modify any logged in user when uid is set
-        # Workaround is to run usermod manually for ansible_ssh_user
-        - user.uid is not defined or (user.uid is defined and user.name != ansible_ssh_user)
+        # Ansible user module and usermod command fail to modify any logged in user when uid is set
+        # Workaround is to run usermod manually for logged in users and not set UID even if it is defined
+        - user.uid is not defined or (user.uid is defined and user.name not in admin_logged_in_users)
```
Comment (oscarkraemer) [inline tasks/main.yml]: Daaaaamn!

File: tasks/main.yml [hunk 3]
```diff
@@ -26,6 +26,16 @@
     when:
       - ansible_ssh_user is undefined
 
+    # This list includes the user used to ssh into the system
+  - name: grab a space separated string of users with the users linux command
+    command: users
+    check_mode: no
+    register: reg_ansible_logged_in_users
+
+  - name: create an ansible list of unique and logged in users
```
Comment (oscarkraemer) [inline tasks/main.yml]: Same here, would it make sense to have a 'changed_when: False' here?

File: tasks/main.yml (hunk 2 above)
Comment (martbhell) [reply tasks/main.yml]: Definitely ;) 
Added change_when here and there and updated some more documentation/styling while at it.
 (#31 was an old PR that hung for a while and eventually conflicted with some other updates. )

**Comment by oscarkraemer** [comment]
The travis test failed. :/

**Comment by martbhell** [comment]
Some pip import error with the pip-ubuntu test only which might be fixed by updating or downgrading pip or setuptools. The other tests pass.

**Comment by oscarkraemer** [comment]
So the test failing is a non-issue in this case?

**Comment by martbhell** [comment]
Yep, the test after this merge passed all the tests :)

**Pull request merged**